tests/arch/arm/arm_runtime_nmi/src/arm_runtime_nmi.c

```diff
@@ -61,6 +61,12 @@ void test_arm_runtime_nmi(void)
 	/* Trigger NMI: Should fire immediately */
 	SCB->ICSR |= SCB_ICSR_NMIPENDSET_Msk;
 
+#ifdef ARM_CACHEL1_ARMV7_H
+	/* Flush Data Cache now if enabled */
+	if (IS_ENABLED(CONFIG_DCACHE)) {
```

Comment:
What about using `#ifdef CONFIG_DCACHE` directly?

Reply:
that's the way soc/arm/st_stm32/stm32h7/soc_m7.c and soc/arm/st_stm32/stm32f7/soc.c do : 
  `if (IS_ENABLED(CONFIG_DCACHE))`  and _not_ `#ifdef CONFIG_DCACHE`